File: 362. Design Hit Counter.py
##ss
class HitCounter:

    def __init__(self):
        self.arr = []
        

    def hit(self, timestamp: int) -> None:
        self.arr.append(timestamp)
        

    def getHits(self, timestamp: int) -> int:
        ans = 0
        for x in range(len(self.arr)-1,-1,-1):
            if timestamp - self.arr[x] < 300:
                ans+=1
                
            else:
                break
                
        return ans
        


# Your HitCounter object will be instantiated and called as such:
# obj = HitCounter()
# obj.hit(timestamp)
# param_2 = obj.getHits(timestamp)

File: cpp_solutions/362. Design Hit Counter.cc
class HitCounter {
public:
    vector<int> arr;
    HitCounter() {
        arr.clear();
    }
    
    void hit(int timestamp) {
        arr.emplace_back(timestamp);
    }
    
    int getHits(int timestamp) {
        int ans = 0;
        for(int x=arr.size()-1; x>=0; --x) {
            if (timestamp - arr[x] < 300) {
                // cout << x << " ";
                ++ans;
            } else {break;}
        }
        // cout << "\n";
        return ans;
    }
};

/**
 * Your HitCounter object will be instantiated and called as such:
 * HitCounter* obj = new HitCounter();
 * obj->hit(timestamp);
 * int param_2 = obj->getHits(timestamp);
 */
